easybuild/tools/config.py

```diff
@@ -170,49 +167,14 @@ def mk_full_default_path(name, prefix=DEFAULT_PREFIX):
 ]
 
 
-OLDSTYLE_ENVIRONMENT_VARIABLES = {
-    'build_path': 'EASYBUILDBUILDPATH',
-    'config_file': 'EASYBUILDCONFIG',
-    'install_path': 'EASYBUILDINSTALLPATH',
-    'log_format': 'EASYBUILDLOGFORMAT',
-    'log_dir': 'EASYBUILDLOGDIR',
-    'source_path': 'EASYBUILDSOURCEPATH',
-    'test_output_path': 'EASYBUILDTESTOUTPUT',
-}
-
-
-OLDSTYLE_NEWSTYLE_MAP = {
-    'build_path': 'buildpath',
-    'install_path': 'installpath',
-    'log_dir': 'tmp_logdir',
-    'config_file': 'config',
-    'source_path': 'sourcepath',
-    'log_format': 'logfile_format',
-    'test_output_path': 'testoutput',
-    'module_classes': 'moduleclasses',
-    'repository_path': 'repositorypath',
-    'modules_install_suffix': 'subdir_modules',
-    'software_install_suffix': 'subdir_software',
-}
-
-
-def map_to_newstyle(adict):
-    """Map a dictionary with oldstyle keys to the new style."""
-    res = {}
-    for key, val in adict.items():
-        if key in OLDSTYLE_NEWSTYLE_MAP:
-            key = OLDSTYLE_NEWSTYLE_MAP[key]
-        res[key] = val
-    return res
-
-
 class ConfigurationVariables(FrozenDictKnownKeys):
     """This is a dict that supports legacy config names transparently."""
 
     # singleton metaclass: only one instance is created
     __metaclass__ = Singleton
 
-    REQUIRED = [
+    # list of know/required keys
```

Comment:
ok, so they might be required :smile: 
just fix the typo, maybe clarify the role of the `no_missing` boolean?